node_modules/gh-orgunit/lib/api.js

```diff
@@ -216,16 +221,15 @@ var getOrgUnits = module.exports.getOrgUnits = function(ctx, appId, parentId, ty
                 return callback(err);
             }
 
-            // If the current user is anonymous or no organisational
-            // units were found, we return them as is
-            if (!ctx.user || _.isEmpty(orgUnits)) {
+            // If we don't need to retrieve whether the current user can manage
+            // the organisational units and series, we can return early
+            if (!includePermissions || !ctx.user) {
                 var augmentedOrgUnits = _augmentOrgUnits(ctx, orgUnits, []);
                 return callback(null, augmentedOrgUnits);
             }
 
             // Otherwise we get the groups the current user is a member of so we can mark
             // those organisational units that the user can edit
-            // TODO: Make this optional
```

Comment:
When I was doing simple benchmarks on my machine it made a pretty big difference. I'm pretty confident this would have a significant impact in a fully scaled-out environment